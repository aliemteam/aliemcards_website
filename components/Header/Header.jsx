import * as React from 'react';
import Link from 'next/link';
import './header.sass';

import Search from '../Search/Search';

class Header extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      navDrawerOpen: false,
    };
  }

  handleClick = () => {
    this.setState({ navDrawerOpen: !this.state.navDrawerOpen });
  };

  render() {
    return (
      <div>
        <div className="header" role="banner">
          <div className="header__logo">
            <Link href="/">
              <a><img src="/static/logo.svg" /></a>
            </Link>
          </div>
          <div className="header__nav-button">
            <button
              className={
                this.state.navDrawerOpen
                  ? 'hamburger hamburger__squeeze hamburger__squeeze--active'
                  : 'hamburger hamburger__squeeze'
              }
              onClick={this.handleClick}
            >
              <span className="hamburger__box">
                <span className="hamburger__inner" />
              </span>
            </button>
          </div>
          <nav
            className={
              this.state.navDrawerOpen
                ? 'header__nav header__nav--open'
                : 'header__nav header__nav--closed'
            }
            role="navigation"
          >
            <ul>
              <li>
                <Link href="/cards">
                  <a onClick={this.handleClick}>Cards</a>
                </Link>
              </li>
              <li>
                <Link href="/categories">
                  <a onClick={this.handleClick}>Categories</a>
                </Link>
              </li>
              <li>
                <Link href="/about">
                    <a onClick={this.handleClick}>About</a>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
        <Search />
      </div>
    );
  }
}

export default Header;
