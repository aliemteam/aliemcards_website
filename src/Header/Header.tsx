import * as React from 'react';
import { Link, withRouter } from 'react-router-dom';

import Search from '../Search/Search';

const logo = require('./logo.svg');
const style = require('./header.sass');

interface State {
  navDrawerOpen: boolean;
}

class Header extends React.PureComponent<{}, State> {
  constructor(props) {
    super(props);
    this.state = {
      navDrawerOpen: false,
    };
  }

  handleClick = () => {
    this.setState({ navDrawerOpen: !this.state.navDrawerOpen });
  };

  render() {
    return (
      <div>
        <div className="header" role="banner">
          <div className="header__logo">
            <Link to="/">
              <img src={logo} />
            </Link>
          </div>
          <div className="header__nav-button">
            <button
              className={
                this.state.navDrawerOpen
                  ? 'hamburger hamburger__squeeze hamburger__squeeze--active'
                  : 'hamburger hamburger__squeeze'
              }
              onClick={this.handleClick}
            >
              <span className="hamburger__box">
                <span className="hamburger__inner" />
              </span>
            </button>
          </div>
          <nav
            className={
              this.state.navDrawerOpen
                ? 'header__nav header__nav--open'
                : 'header__nav header__nav--closed'
            }
            role="navigation"
          >
            <ul>
              <li>
                <Link to="/cards" onClick={this.handleClick}>
                  Cards
                </Link>
              </li>
              <li>
                <Link to="/categories" onClick={this.handleClick}>
                  Categories
                </Link>
              </li>
              <li>
                <Link to="/about" onClick={this.handleClick}>
                  About
                </Link>
              </li>
            </ul>
          </nav>
        </div>
        <Search />
      </div>
    );
  }
}

export default withRouter(Header);
