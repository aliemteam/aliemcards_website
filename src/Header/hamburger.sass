.hamburger
    padding: 8px
    display: inline-block
    cursor: pointer
    transition: opacity linear .15s, filter linear .15s
    font: inherit
    color: inherit
    text-transform: none
    background-color: transparent
    border: 0
    margin: 0
    overflow: visible
    &:hover
        opacity: .7

.hamburger__box
    width: 30px
    height: 24px
    display: inline-block
    position: relative

.hamburger__inner
    display: block
    top: 50%
    margin-top: -2px
    &,
    &::before,
    &::after
        width: 30px
        height: 3px
        background-color: $dark-gray
        border-radius: 4px
        position: absolute
        transition-property: transform
        transition-duration: .15s
        transition-timing-function: ease
    &::before,
    &::after
        content: ''
        display: block
    &::before
        top: -10px
    &::after
        bottom: -10px

.hamburger__squeeze
    .hamburger__inner
        transition-duration: .1s
        transition-timing-function: cubic-bezier(.55, .055, .675, .19)
        &::before
            transition: top .1s .14s ease, opacity .1s ease
        &::after
            transition: bottom .1s .14s ease, transform .1s cubic-bezier(.55, .055, .675, .19)

.hamburger__squeeze--active
    .hamburger__inner
        transform: rotate(45deg)
        transition-delay: .14s
        transition-timing-function: cubic-bezier(.215, .61, .355, 1)
        &::before
            top: 0
            opacity: 0
            transition: top .1s ease, opacity .1s .14s ease
        &::after
            bottom: 0
            transform: rotate(-90deg)
            transition: bottom .1s ease, transform .1s .14s cubic-bezier(.215, .61, .355, 1)
