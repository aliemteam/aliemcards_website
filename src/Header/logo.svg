<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0" y="0" width="247.605" height="35" viewBox="0, 0, 247.605, 35">
  <g id="aliem-cards-logo">
    <path d="M6.604,21.626 C7.009,21.076 7.446,20.522 7.908,19.966 C7.733,19.632 7.632,19.252 7.63,18.848 C4.799,18.231 2.168,18.34 0,18.711 L0,23.451 L5.341,23.451 C5.738,22.842 6.156,22.233 6.604,21.626" fill="#00B092" id="path3448"/>
    <path d="M12.481,18.794 C12.924,19.005 13.361,19.23 13.793,19.472 C14.227,19.062 14.811,18.81 15.456,18.81 C16.224,18.81 16.907,19.167 17.352,19.724 C18.555,19.356 19.794,19.074 21.065,18.908 C21.674,18.829 22.278,18.784 22.88,18.759 C23.001,18.412 23.196,18.099 23.45,17.843 C22.788,16.329 22.01,14.894 21.113,13.541 C20.844,13.643 20.554,13.703 20.249,13.703 C19.653,13.703 19.107,13.486 18.685,13.13 C16.348,14.39 14.195,15.934 12.227,17.759 C12.384,18.072 12.474,18.423 12.481,18.794" fill="#00B092" id="path3450"/>
    <path d="M13.028,21.236 C13.028,21.132 13.037,21.029 13.05,20.927 C12.688,20.723 12.322,20.533 11.953,20.352 C11.508,20.909 10.824,21.266 10.057,21.266 C9.721,21.266 9.4,21.198 9.109,21.074 C8.703,21.568 8.304,22.072 7.917,22.595 C7.708,22.88 7.506,23.165 7.308,23.45 L9.62,23.45 C10.702,22.739 11.844,22.069 13.04,21.468 C13.033,21.391 13.028,21.314 13.028,21.236" fill="#00B092" id="path3452"/>
    <path d="M23.803,29.351 L23.803,23.45 L25.272,23.45 C25.157,22.948 25.034,22.452 24.898,21.964 C23.973,21.861 23.207,21.237 22.896,20.393 C22.358,20.415 21.818,20.456 21.276,20.526 C20.118,20.677 18.984,20.934 17.88,21.272 C17.875,21.603 17.805,21.917 17.681,22.203 C19.896,24.1 21.939,26.486 23.803,29.351" fill="#00B092" id="path3454"/>
    <path d="M16.564,23.394 C16.231,23.565 15.855,23.663 15.455,23.663 C14.788,23.663 14.183,23.394 13.745,22.957 C13.115,23.277 12.5,23.617 11.901,23.973 L11.901,35 L23.803,35 L23.803,32.489 C21.652,28.703 19.235,25.666 16.564,23.394" fill="#00B092" id="path3456"/>
    <path d="M18.119,7.47 C18.596,7.921 19.082,8.414 19.569,8.947 C19.785,8.884 20.013,8.849 20.249,8.849 C21.045,8.849 21.75,9.234 22.192,9.827 C22.716,9.631 23.254,9.447 23.802,9.27 L23.802,-0 L11.901,-0 L11.901,2.944 C13.746,3.953 15.943,5.411 18.119,7.47" fill="#00B092" id="path3458"/>
    <path d="M27.278,20.754 C27.081,21.1 26.802,21.391 26.466,21.604 C26.632,22.202 26.786,22.817 26.927,23.45 L33.752,23.45 C31.66,22.121 29.498,21.221 27.278,20.754" fill="#00B092" id="path3460"/>
    <path d="M8.182,17.298 C8.627,16.757 9.302,16.413 10.057,16.413 C10.42,16.413 10.763,16.495 11.072,16.637 C12.98,14.88 15.234,13.215 17.877,11.788 C17.842,11.623 17.822,11.452 17.822,11.276 C17.822,10.77 17.977,10.301 18.242,9.913 C17.841,9.484 17.428,9.063 16.998,8.656 C15.247,7 13.468,5.751 11.901,4.829 L11.901,11.55 L-0,11.55 L-0,17.042 C2.337,16.67 5.147,16.599 8.182,17.298" fill="#00B092" id="path3462"/>
    <path d="M23.803,11.549 L23.803,10.917 C23.423,11.045 23.046,11.178 22.674,11.318 C22.667,11.747 22.547,12.148 22.345,12.494 C23.257,13.864 24.12,15.409 24.883,17.145 C24.978,17.134 25.073,17.127 25.171,17.127 C26.37,17.127 27.363,17.997 27.56,19.14 C30.372,19.722 33.094,20.943 35.704,22.795 L35.704,11.549 L23.803,11.549" fill="#00B092" id="path3464"/>
    <path d="M57.162,11.505 L54.108,19.315 L60.25,19.315 L57.162,11.505 L57.162,11.505 z M55.032,4.191 L59.611,4.191 L70.545,29.326 L64.297,29.326 L62.132,24.001 L52.369,24.001 L50.274,29.326 L44.169,29.326 L55.032,4.191" fill="#706D6E" id="path3466"/>
    <path d="M73.279,4.191 L78.817,4.191 L78.817,24.214 L89.076,24.214 L89.076,29.326 L73.279,29.326 L73.279,4.191" fill="#706D6E" id="path3468"/>
    <path d="M91.526,12.072 L96.85,12.072 L96.85,29.326 L91.526,29.326 L91.526,12.072 z M91.099,6.641 C91.099,5.789 91.401,5.061 92.005,4.458 C92.608,3.854 93.335,3.553 94.188,3.553 C95.04,3.553 95.768,3.855 96.371,4.458 C96.975,5.061 97.276,5.789 97.276,6.641 C97.276,7.493 96.975,8.221 96.371,8.824 C95.768,9.428 95.04,9.729 94.188,9.729 C93.336,9.729 92.608,9.428 92.005,8.824 C91.401,8.221 91.099,7.493 91.099,6.641" fill="#706D6E" id="path3470"/>
    <path d="M101.572,4.191 L118.647,4.191 L118.647,9.303 L107.11,9.303 L107.11,13.989 L118.009,13.989 L118.009,19.101 L107.11,19.101 L107.11,24.214 L119.287,24.214 L119.287,29.326 L101.572,29.326 L101.572,4.191" fill="#706D6E" id="path3472"/>
    <path d="M123.724,4.191 L132.102,4.191 L137.888,20.592 L137.959,20.592 L143.781,4.191 L152.124,4.191 L152.124,29.326 L146.586,29.326 L146.586,10.049 L146.515,10.049 L139.912,29.326 L135.687,29.326 L129.332,10.049 L129.262,10.049 L129.262,29.326 L123.723,29.326 L123.723,4.191" fill="#706D6E" id="path3474"/>
    <g>
      <path d="M172.686,6.823 C169.761,6.823 167.459,7.758 165.78,9.63 C164.1,11.501 163.26,14.064 163.26,17.317 C163.26,20.645 164.052,23.229 165.636,25.068 C167.22,26.908 169.485,27.828 172.43,27.828 C174.386,27.828 176.183,27.577 177.82,27.078 L177.82,28.513 C176.279,29.055 174.354,29.326 172.047,29.326 C168.772,29.326 166.194,28.268 164.312,26.152 C162.43,24.036 161.489,21.08 161.489,17.285 C161.489,14.914 161.939,12.824 162.837,11.017 C163.735,9.209 165.027,7.814 166.713,6.831 C168.397,5.847 170.357,5.355 172.59,5.355 C174.865,5.355 176.901,5.781 178.698,6.631 L178.044,8.098 C176.343,7.248 174.557,6.823 172.686,6.823 z" fill="#706D6E"/>
      <path d="M193.147,29.007 L192.748,26.264 L192.62,26.264 C191.749,27.381 190.853,28.17 189.934,28.633 C189.014,29.095 187.927,29.327 186.672,29.327 C184.971,29.327 183.647,28.89 182.701,28.018 C181.755,27.147 181.281,25.94 181.281,24.398 C181.281,22.708 181.986,21.394 183.395,20.459 C184.804,19.523 186.842,19.029 189.511,18.976 L192.812,18.88 L192.812,17.731 C192.812,16.084 192.477,14.84 191.808,14 C191.137,13.16 190.059,12.74 188.57,12.74 C186.964,12.74 185.3,13.186 183.578,14.08 L182.988,12.708 C184.891,11.815 186.773,11.369 188.634,11.369 C190.536,11.369 191.959,11.863 192.9,12.852 C193.841,13.84 194.311,15.382 194.311,17.476 L194.311,29.007 L193.147,29.007 z M186.768,27.891 C188.618,27.891 190.077,27.362 191.146,26.304 C192.214,25.246 192.748,23.776 192.748,21.894 L192.748,20.188 L189.718,20.316 C187.284,20.432 185.548,20.813 184.511,21.455 C183.474,22.099 182.956,23.101 182.956,24.462 C182.956,25.546 183.289,26.389 183.953,26.99 C184.617,27.59 185.556,27.891 186.768,27.891 z" fill="#706D6E"/>
      <path d="M206.751,11.336 C207.484,11.336 208.271,11.411 209.111,11.559 L208.808,13.075 C208.084,12.894 207.335,12.804 206.559,12.804 C205.081,12.804 203.869,13.431 202.923,14.686 C201.976,15.941 201.503,17.524 201.503,19.438 L201.503,29.007 L199.924,29.007 L199.924,11.671 L201.264,11.671 L201.423,14.797 L201.535,14.797 C202.248,13.521 203.008,12.626 203.816,12.11 C204.624,11.594 205.603,11.336 206.751,11.336 z" fill="#706D6E"/>
      <path d="M218.983,11.336 C220.238,11.336 221.322,11.565 222.237,12.022 C223.151,12.479 223.97,13.261 224.692,14.366 L224.788,14.366 C224.725,13.027 224.692,11.714 224.692,10.427 L224.692,4.191 L226.256,4.191 L226.256,29.007 L225.219,29.007 L224.82,26.36 L224.692,26.36 C223.374,28.338 221.481,29.327 219.015,29.327 C216.622,29.327 214.793,28.582 213.528,27.094 C212.263,25.605 211.631,23.436 211.631,20.586 C211.631,17.588 212.258,15.297 213.512,13.712 C214.767,12.128 216.591,11.336 218.983,11.336 z M218.983,12.772 C217.091,12.772 215.668,13.436 214.717,14.765 C213.765,16.094 213.29,18.024 213.29,20.554 C213.29,25.466 215.199,27.922 219.016,27.922 C220.971,27.922 222.407,27.354 223.321,26.216 C224.236,25.079 224.692,23.202 224.692,20.586 L224.692,20.315 C224.692,17.636 224.243,15.709 223.345,14.534 C222.447,13.359 220.992,12.772 218.983,12.772 z" fill="#706D6E"/>
      <path d="M242.826,24.494 C242.826,26.046 242.236,27.24 241.056,28.074 C239.876,28.909 238.201,29.327 236.032,29.327 C233.714,29.327 231.875,28.97 230.514,28.258 L230.514,26.551 C232.258,27.423 234.097,27.859 236.032,27.859 C237.744,27.859 239.044,27.575 239.931,27.006 C240.819,26.437 241.263,25.68 241.263,24.734 C241.263,23.861 240.909,23.128 240.202,22.532 C239.495,21.937 238.333,21.353 236.717,20.778 C234.985,20.151 233.767,19.611 233.065,19.159 C232.364,18.708 231.834,18.197 231.479,17.628 C231.122,17.06 230.944,16.366 230.944,15.547 C230.944,14.25 231.489,13.224 232.579,12.469 C233.668,11.714 235.192,11.337 237.148,11.337 C239.019,11.337 240.795,11.687 242.474,12.389 L241.884,13.824 C240.183,13.123 238.604,12.772 237.148,12.772 C235.733,12.772 234.612,13.006 233.782,13.473 C232.953,13.941 232.539,14.59 232.539,15.419 C232.539,16.323 232.86,17.045 233.503,17.588 C234.147,18.131 235.404,18.736 237.275,19.406 C238.838,19.97 239.975,20.477 240.688,20.93 C241.4,21.382 241.934,21.894 242.291,22.469 C242.648,23.042 242.826,23.718 242.826,24.494 z" fill="#706D6E"/>
    </g>
  </g>
</svg>
