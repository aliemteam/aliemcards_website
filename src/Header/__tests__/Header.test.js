import React from 'react';
import Header from '../Header';
import ShallowRenderer from 'react-test-renderer/shallow';



it('renders correctly', () => {
  const renderer = new ShallowRenderer;
  const tree = renderer.render(<Header />);
  expect(tree).toMatchSnapshot();
});
