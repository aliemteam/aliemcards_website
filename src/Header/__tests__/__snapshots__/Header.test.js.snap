// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
    role="banner"
  >
    <div
      className="header__logo"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          src="logo.svg"
        />
      </Link>
    </div>
    <div
      className="header__nav-button"
    >
      <button
        className="hamburger hamburger__squeeze"
        onClick={[Function]}
      >
        <span
          className="hamburger__box"
        >
          <span
            className="hamburger__inner"
          />
        </span>
      </button>
    </div>
    <nav
      className="header__nav header__nav--closed"
      role="navigation"
    >
      <ul>
        <li>
          <Link
            onClick={[Function]}
            replace={false}
            to="/cards"
          >
            Cards
          </Link>
        </li>
        <li>
          <Link
            onClick={[Function]}
            replace={false}
            to="/categories"
          >
            Categories
          </Link>
        </li>
        <li>
          <Link
            onClick={[Function]}
            replace={false}
            to="/about"
          >
            About
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
